speechbrain/decoders/seq2seq.py

```diff
@@ -810,7 +810,7 @@ def forward(self, enc_states, wav_len):  # noqa: C901
             log_probs,
         ) = self._get_top_score_prediction(hyps_and_scores, topk=self.topk,)
         # pick the best hyp
-        predictions = topk_hyps[:, 0, :]
+        predictions = topk_hyps[:, :self.topk, :]
```

Comment:
@Gastron , yes, returning the top-k hypotheses will take extra dimension. 
The shape of `topk_hyps` is `[B, topk, L]`, we pick the top hyps (`predictions: [B, L]`) and pass it to `batch_filter_seq2seq_output` to filter zeros. This way we return a list of top hyps not tensor.
Maybe a simpler way is returning `predictions` and `topk_hyps`,  `topk_lens`.

Also this line is redundant, the shape of `topk_hyps` is already `[B, topk, L]`.
```
predictions = topk_hyps[:, :self.topk, :]
```

Reply:
Oh ok. If we change the return, then doesn't that also involve changing anything that uses this class?

Reply:
It does if we want to return topk hyps.

Reply:
All the current recipes use only top hyps for validation/testing.